internal/kafka/internal/handlers/validation.go

```diff
@@ -51,46 +51,109 @@ func ValidateKafkaClusterNameIsUnique(name *string, kafkaService services.KafkaS
 	}
 }
 
-// ValidateCloudProvider returns a validator that sets default cloud provider details if needed and validates provided
-// provider and region
-func ValidateCloudProvider(kafkaService *services.KafkaService, kafkaRequest *dbapi.KafkaRequest, providerConfig *config.ProviderConfig, action string) handlers.Validate {
-	return func() *errors.ServiceError {
-		// Set Cloud Provider default if not received in the request
-		supportedProviders := providerConfig.ProvidersConfig.SupportedProviders
-		if kafkaRequest.CloudProvider == "" {
-			defaultProvider, _ := supportedProviders.GetDefault()
-			kafkaRequest.CloudProvider = defaultProvider.Name
-		}
+func getCloudProviderAndRegion(
```

Comment:
Just as a thought, do you think this could be further split?

Quickly looking at it, it seems it validates the value if it is provided. If it is not provided then a default is assigned.

We could have a method that only validates, and then on the action handler it could be checked whether the attribute is defined. If it is defined we know it is valid so we can just assign it. If it is not defined we know we should set the default value.
In this way we would avoid repeating the validation. It might not be worth it though

Reply:
I didn't mean splitting the region and provider logic, but splitting the validation from the retrieval